src/Sylius/Bundle/UserBundle/Security/UserPasswordHasher.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\UserBundle\Security;
+
+use Sylius\Component\User\Model\CredentialsHolderInterface;
+use Sylius\Component\User\Security\UserPasswordEncoderInterface;
+use Sylius\Component\User\Security\UserPasswordHasherInterface;
+use Symfony\Component\PasswordHasher\Hasher\PasswordHasherFactoryInterface;
+
+final class UserPasswordHasher implements UserPasswordEncoderInterface
+{
+    public function __construct(private PasswordHasherFactoryInterface $passwordHasherFactory)
+    {
+    }
+
+    public function encode(CredentialsHolderInterface $user): string
+    {
+        return $this->hash($user);
+    }
+
+    public function hash(CredentialsHolderInterface $user): string
```

Comment:
This method is not part of any contract, let's make it private (or introduce a second interface here, implement both of them in this class and eventually we will change the contract with Sylius 2.0, but I would vote for making this function just private)
```suggestion
    private function hash(CredentialsHolderInterface $user): string
```